nautobot_netbox_importer/diffsync/models/dcim.py

```diff
@@ -308,6 +311,23 @@ class Interface(BaseInterfaceMixin, CableTerminationMixin, ComponentModel):
     untagged_vlan: Optional[VLANRef]
     tagged_vlans: List[VLANRef] = []
 
+    @root_validator
+    def invalid_type_to_other(cls, values):
+        int_type = values.get("type")
+        if int_type and int_type not in INTERFACE_TYPE_CHOICES:
+            values["type"] = "other"
+            int_name = values.get("name", "Unknonw")
+            int_device = values.get("device", "Unknown")
+            if int_device != "Unknown":
+                device = dcim.Device.objects.filter(pk=int_device)
```

Comment:
It will have the Device UUID at this point, but that Device may or may not already be in the database. The first time loading the device and interface would not have the Device created, but afterwards it would:

```
15:30:28 warning  Netbox Interface.type of DNE is not supported by this version of Nautobot. 
Interface, mgmt10, on Device, 75e34896-d2f5-5ed1-85a7-c17ecbd0d64a, will use a type of 'other'

15:32:48 warning  Netbox Interface.type of DNE is not supported by this version of Nautobot. 
Interface, mgmt10, on Device, VC A Member 2, will use a type of 'other'
```